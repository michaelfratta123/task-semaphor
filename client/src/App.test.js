import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import App from "./App";
import LoginRegister from "./components/LoginRegister";

test("App component renders without crashing", () => {
  const { asFragment } = render(
    <Router>
      <App />
    </Router>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("LoginRegister component is present when there is no userData", () => {
  // Render the LoginRegister component without providing userData
  render(<LoginRegister handleLogin={() => {}} handleRegister={() => {}} />);

  // Check if the LoginRegister component is rendered by querying its data-testid
  const loginRegisterCard = screen.queryByTestId("login-register-card");

  // Assert that the LoginRegister component is present
  expect(loginRegisterCard).toBeInTheDocument();
});
