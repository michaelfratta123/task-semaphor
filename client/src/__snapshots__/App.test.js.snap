// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="bg-body-tertiary navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container"
      >
        <span
          class="AppTitle fs-1 navbar-brand"
        >
          TaskSemaphor
        </span>
        <div
          class="navbar-nav"
        />
      </div>
    </nav>
    <div>
      <div
        class="m-auto text-light mt-5 card bg-dark border-light"
        data-testid="login-register-card"
        style="width: 20rem;"
      >
        <div
          class="card-body"
        >
          <div
            class="mb-4 mt-2 card-title h5"
          >
            Login or Register
          </div>
          <form
            class=""
          >
            <div
              class="mb-3 mx-4"
            >
              <input
                class="form-control"
                id="formBasicUsername"
                name="username"
                placeholder="Username"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-3 mx-4"
            >
              <input
                class="form-control"
                id="formBasicPassword"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="m-3 btn btn-secondary"
              type="button"
            >
              Login
            </button>
            <button
              class="btn btn-secondary"
              type="button"
            >
              Register
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
